src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts

```diff
@@ -38,8 +43,21 @@ export const useTradeExecution = ({
     selectPortfolioAccountMetadataByAccountId(state, { accountId: sellAssetAccountId }),
   )
 
+  const buyAssetUsdRate = useAppSelector(state =>
+    selectUsdRateByAssetId(state, tradeQuote?.steps[0].buyAsset.assetId ?? ('' as AssetId)),
+  )
+
+  const feeAsset = useAppSelector(state =>
+    selectFeeAssetById(state, tradeQuote?.steps[0].sellAsset.assetId ?? ''),
+  )
```

Comment:
```suggestion
  const tradeQuoteStep = useMemo(() => tradeQuote?.steps[0], [tradeQuote])
  const buyAssetUsdRate = useAppSelector(state =>
    tradeQuoteStep ? selectUsdRateByAssetId(state, tradeQuoteStep.buyAsset.assetId) : undefined,
  )

  const feeAsset = useAppSelector(state =>
    tradeQuoteStep ? selectFeeAssetById(state, tradeQuoteStep.sellAsset.assetId) : undefined,
  )
```

Reply:
See https://github.com/shapeshift/web/pull/4847/commits/c74339b12b9b75e9f5930866150318945d88049c for a version using the proper selectors.